rust/src/builders/witness_builder.rs

```diff
@@ -303,6 +302,58 @@ impl TransactionWitnessSetBuilder {
         };
     }
 
+    fn add_fake_vkey_witnesses(&mut self, vkeys: &Vec<Vkey>) {
+        let fake_sig = fake_raw_key_sig(0);
+        for vkey in vkeys {
+            let fake_vkey_witness = Vkeywitness::new(&vkey, &fake_sig);
+            self.add_vkey(&fake_vkey_witness);
+        }
+    }
+
+    fn add_fake_vkey_witnesses_by_num(&mut self, num: usize) {
+        let vkeys: Vec<Vkey> = (0..num).into_iter().map(|i| Vkey::new(&fake_raw_key_public((i + self.vkeys.len()) as u8))).collect();
+        self.add_fake_vkey_witnesses(&vkeys);
+    }
+
+    pub fn add_input_aggregate_witness_data(&mut self, data: &InputAggregateWitnessData) {
```

Comment:
This function won't work in WASM because this struct (`TransactionWitnessSetBuilder`) is visible for wasm, but this member function has an argument `InputAggregateWitnessData` which isn't visible from WASM (and note that this enum can't easily be exposed to WASM either)

Since this function never needs to be called from WASM, you may want to just make it so it's not a member function (just place it somewhere else in the file)

The proper way to fix this would have been to make that all of `TransactionWitnessSetBuilder` is a non-WASM Rust-only struct (and therefore supports this function) and then expose a WASM-compatible subset of the struct

Reply:
How about making the function public only in this crate, like those attributes in other builders. This function shouldn't be visible anyway.

Reply:
Updated.